src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -6671,7 +6677,7 @@ public static MethodHandle loop(MethodHandle[]... clauses) {
                 pred.set(i, dropArguments0(constant(boolean.class, true), 0, commonParameterSequence));
             }
             if (fini.get(i) == null) {
-                fini.set(i, empty(methodType(t, commonParameterSequence)));
+                fini.set(i, empty(methodType(loopReturnType, commonParameterSequence)));
```

Comment:
As per https://github.com/eclipse-openj9/openj9/issues/11937#issuecomment-856086725
> We will need to raise an OpenJDK bug report to request them to fix the issue to macth the expected behavior in the Spec.

fyi @andrew-m-leonard

Reply:
A request has been made to open a bug report